easybuild/easyconfigs/j/Julia/Julia-1.5.1-foss-2020a.eb

```diff
@@ -0,0 +1,79 @@
+easyblock = 'ConfigureMake'
+
+name = 'Julia'
+version = '1.5.1'
+
+homepage = 'https://julialang.org'
+description = "Julia is a high-level, high-performance dynamic language for technical computing."
+
+toolchain = {'name': 'foss', 'version': '2020a'}
+toolchainopts = {'rpath': True}
+
+source_urls = ['https://github.com/JuliaLang/julia/releases/download/v%(version)s/']
+sources = ['julia-%(version)s-full.tar.gz']
+checksums = ['1d0debfccfc7cd07047aa862dd2b1a96f7438932da1f5feff6c1033a63f9b1d4']
+
+builddependencies = [
+    ('Autotools', '20180311'),
+    ('CMake', '3.16.4'),
+    ('Python', '3.8.2'),
+    ('Perl', '5.30.2'),
+    ('pkg-config', '0.29.2'),
+    ('PCRE2', '10.34'),
+    ('patchelf', '0.12'),
+]
+
+# some dependencies are provided through EasyBuild,
+# but for others we let Julia build the ones included in the sources;
+# adding SuiteSparse as a proper dep leads to problems with the tests,
+# because some fail because they unset $LD_LIBRARY_PATH which leads to
+# libraries like libamd.so not being found
+dependencies = [
+    ('GMP', '6.2.0'),
+    ('MPFR', '4.0.2'),
+    ('cURL', '7.69.1'),
+    ('zlib', '1.2.11'),
+    ('SuiteSparse', '5.7.1', '-METIS-5.1.0'),
+]
```

Comment:
Should we list deps we leave to Julia to do it's own thing with? Based on my incomplete, broken, attempt, I identified:
LLVM
OpenLibm
utf8proc
libuv
libgit2
p7zip
dSFMT

Reply:
We could, see https://github.com/JuliaLang/julia/blob/master/doc/build/build.md#required-build-tools-and-external-libraries (some are only needed on Windows)

Reply:
We could probably could use these external as well if we want:
* OpenLibm & utf8proc: No patches  (though this is more or less just Julia specific anyway)
* p7zip: Patches seem to CVE related, nothing julia specific
* libunwind: Patches seem to be ARM specific
* libgit2: Required patch to make ssh-agent problems non-fatal. Probably required, but I suppose it wouldn't hurt elsewhere, so we could possibly include that. It's been proposed for inclusion upstream (but without response from git devs).

Note: As Julia is transitioning more towards their BinaryBuilder tool where Julia modules can depend on pre-built "Artifacts", called "JLL"s, there is some potential conflict there if we try to control the versions to much and pulling in other libraries. Hard to say how it's all going to pan out.

Reply:
@Micket "libunwind: Patches seem to be ARM specific" is interesting. Could you expand on this?
We see Mesa and probably more stuff fail on arm due to libunwind: https://github.com/easybuilders/easybuild-easyblocks/pull/2153. Is libunwind on arm something we can fix? I see arch and others ship it and use it for Mesa: https://archlinuxarm.org/packages/aarch64/mesa

Reply:
@terjekv This comment was just based on the patchset i found in julia: https://github.com/JuliaLang/julia/blob/master/deps/patches/ (loooks like a very very minor patch though). Also the yggdrasil package stuff: https://github.com/JuliaPackaging/Yggdrasil/tree/master/L/LibUnwind/bundled/patches. That's all i know.